SU-HSDO/suhumsci
# STN-157 : Date Stacked Horizontal UI Pattern

# [STN-157](https://sparkbox.atlassian.net/browse/STN-157) READY FOR REVIEW

## Summary
This PR creates a `Date Stacked Horizontal` card designed to be used specifically for events.

This PR has been browser tested ✅, accessibility tested ✅, and reviewed by a designer ✅.

### 📝FYI 
The label style for the Date Stacked Horizontal UI Pattern is under design consideration by H&S. There may be an update to the color of the label.

## Need Review By (Date)
4/22/2020

## Urgency
medium

## Steps to Test
1. In the CLI, run `npm run test` and confirm our Sass tests pass.
1. Because this PR brings in a new pattern, you need to clear the cache by running `lando drush @economics.local cr`. This allows Drupal to recognize the new pattern.

### Add a new view
1. Go to the H&S/Sparkbox shared Google Drive Development directory and download an updated version of the `Custom Econ Events` view.
1. Go to Configuration > Development > Configuration Synchronization > Import
1. Click on the `Single item` tab, under `Configuration type` select `View`, paste settings from `custom-econ-events.txt`, and click `Import`.

### Create a basic page with the new view set to use the Date Stacked Horizontal pattern
1. Create a `Basic Page`
    * Content > Add Content > Basic Page
1. Add a `View` set to `Custom Econ Events` with a `Display` of `Date Horizontal - Single`
1. Save and check out your new page!

### Confirm that you can change the label styles
1. Edit the `Custom Econ Events` view `Date Horizontal - Single` display.
1. Under `Fields`, click on `Content: Speaker (Speaker)`
1. Under `Style Settings`, check `Customize field and label wrapper HTML`
1. Check `Create a CSS class` and add `hb-highlighted-label`
1. Apply and Save


## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
